Mathlib/MeasureTheory/Measure/FiniteMeasure.lean

```diff
@@ -762,12 +762,95 @@ theorem tendsto_iff_forall_integral_tendsto {γ : Type*} {F : Filter γ} {μs :
 
-end FiniteMeasureConvergenceByBoundedContinuousFunctions
-
--- section
-end FiniteMeasure
-
--- namespace
-end MeasureTheory
-
--- namespace
+end FiniteMeasureConvergenceByBoundedContinuousFunctions -- section
+
+section map
+
+variable {Ω Ω' : Type _} [MeasurableSpace Ω] [MeasurableSpace Ω']
+
+/-- The push-forward of a finite measure by a function between measurable spaces. -/
+noncomputable def map (ν : FiniteMeasure Ω) (f : Ω → Ω') : FiniteMeasure Ω' :=
+  ⟨(ν : Measure Ω).map f, by
+    refine ⟨?_⟩
+    by_cases f_aemble : AEMeasurable f ν
+    · rw [Measure.map_apply_of_aemeasurable f_aemble MeasurableSet.univ]
+      exact measure_lt_top (↑ν) (f ⁻¹' univ)
+    · simp [Measure.map, f_aemble]⟩
+
+--#check Subtype.map
+-- Q: Can I tell Lean not to use `Subtype.map` but instead `FiniteMeasure.map`
+-- when `ν.map` is written?
+lemma map_apply' (ν : FiniteMeasure Ω) {f : Ω → Ω'} (f_aemble : AEMeasurable f ν)
+    {A : Set Ω'} (A_mble : MeasurableSet A) :
+    (FiniteMeasure.map ν f : Measure Ω') A = (ν : Measure Ω) (f ⁻¹' A) :=
+  Measure.map_apply_of_aemeasurable f_aemble A_mble
+
+lemma map_apply_of_aemeasurable (ν : FiniteMeasure Ω) {f : Ω → Ω'} (f_aemble : AEMeasurable f ν)
+    {A : Set Ω'} (A_mble : MeasurableSet A) :
+    (FiniteMeasure.map ν f) A = ν (f ⁻¹' A) := by
+  have key := FiniteMeasure.map_apply' ν f_aemble A_mble
+  exact (ENNReal.toNNReal_eq_toNNReal_iff' (measure_ne_top _ _) (measure_ne_top _ _)).mpr key
+
+@[simp] lemma map_apply (ν : FiniteMeasure Ω) {f : Ω → Ω'} (f_mble : Measurable f)
+    {A : Set Ω'} (A_mble : MeasurableSet A) :
+    (FiniteMeasure.map ν f) A = ν (f ⁻¹' A) :=
+  map_apply_of_aemeasurable ν f_mble.aemeasurable A_mble
+
+@[simp] lemma map_add {f : Ω → Ω'} (f_mble : Measurable f) (ν₁ ν₂ : FiniteMeasure Ω)  :
+    FiniteMeasure.map (ν₁ + ν₂) f = FiniteMeasure.map ν₁ f + FiniteMeasure.map ν₂ f := by
+  ext s s_mble
+  simp [map_apply' _ f_mble.aemeasurable s_mble, toMeasure_add]
+
+@[simp] lemma map_smul {f : Ω → Ω'} (f_mble : Measurable f) (c : ℝ≥0) (ν : FiniteMeasure Ω)  :
+    FiniteMeasure.map (c • ν) f = c • FiniteMeasure.map ν f := by
+  ext s s_mble
+  simp [map_apply' _ f_mble.aemeasurable s_mble, toMeasure_smul]
+
+-- Q: Naming?
+/-- The push-forward of a finite measure by a function between measurable spaces as a linear map. -/
+noncomputable def mapHom {f : Ω → Ω'} (f_mble : Measurable f) :
+    FiniteMeasure Ω →ₗ[ℝ≥0] FiniteMeasure Ω' where
+  toFun := fun ν ↦ FiniteMeasure.map ν f
+  map_add' := map_add f_mble
+  map_smul' := map_smul f_mble
+
+variable [TopologicalSpace Ω] [OpensMeasurableSpace Ω]
+variable [TopologicalSpace Ω'] [BorelSpace Ω']
+
+/-- If `f : X → Y` is continuous and `Y` is equipped with the Borel sigma algebra, then
+(weak) convergence of `FiniteMeasure`s on `X` implies (weak) convergence of the push-forwards
+of these measures by `f`. -/
+lemma tendsto_map_of_tendsto_of_continuous {L : Filter ι}
```

Comment:
We have disabled `autoImplicit` in master since this PR was authored so you must now specify `ι`:
```suggestion
lemma tendsto_map_of_tendsto_of_continuous {ι : Type*} {L : Filter ι}
```